x/lockup/keeper/lock.go

```diff
@@ -125,553 +62,548 @@ func (k Keeper) AddToExistingLock(ctx sdk.Context, owner sdk.AccAddress, coin sd
-// This also saves the lock to the store.
+// AddTokensToLock locks additional tokens into an existing lock with the given ID.
+// Tokens locked are sent and kept in the module account.
+// This method alters the lock state in store, thus we do a sanity check to ensure
+// lock owner matches the given owner.
 func (k Keeper) AddTokensToLockByID(ctx sdk.Context, lockID uint64, owner sdk.AccAddress, coin sdk.Coin) (*types.PeriodLock, error) {
 	lock, err := k.GetLockByID(ctx, lockID)
+	if err != nil {
+		return nil, err
+	}
 
 	if lock.GetOwner() != owner.String() {
 		return nil, types.ErrNotLockOwner
 	}
 
+	lock.Coins = lock.Coins.Add(coin)
+	err = k.Lock(ctx, *lock)
 	if err != nil {
 		return nil, err
 	}
-	if err := k.bk.SendCoinsFromAccountToModule(ctx, lock.OwnerAddress(), types.ModuleName, sdk.NewCoins(coin)); err != nil {
-		return nil, err
-	}
 
-	err = k.addTokenToLock(ctx, lock, coin)
-	if err != nil {
-		return nil, err
+	// accumulation store for its synthetic denom is increased if exists.
+	lockedCoin, err := lock.SingleCoin()
+	if err == nil {
+		for _, synthlock := range k.GetAllSyntheticLockupsByLockup(ctx, lock.ID) {
+			k.accumulationStore(ctx, synthlock.SynthDenom).Increase(accumulationKey(synthlock.Duration), sdk.NewCoins(coin).AmountOf(lockedCoin.Denom))
+		}
 	}
 
 	if k.hooks == nil {
 		return lock, nil
 	}
+	k.hooks.AfterAddTokensToLock(ctx, lock.OwnerAddress(), lock.GetID(), sdk.NewCoins(coin))
 
-	k.hooks.OnTokenLocked(ctx, lock.OwnerAddress(), lock.ID, sdk.Coins{coin}, lock.Duration, lock.EndTime)
 	return lock, nil
 }
 
-// SlashTokensFromLockByID send slashed tokens to community pool - called by superfluid module only.
-func (k Keeper) SlashTokensFromLockByID(ctx sdk.Context, lockID uint64, coins sdk.Coins) (*types.PeriodLock, error) {
-	lock, err := k.GetLockByID(ctx, lockID)
+// CreateLock creates a new lock with the specified duration for the owner.
+func (k Keeper) CreateLock(ctx sdk.Context, owner sdk.AccAddress, coins sdk.Coins, duration time.Duration) (types.PeriodLock, error) {
+	ID := k.GetLastLockID(ctx) + 1
+	// unlock time is initially set without a value, gets set as unlock start time + duration
+	// when unlocking starts.
+	lock := types.NewPeriodLock(ID, owner, duration, time.Time{}, coins)
+	err := k.Lock(ctx, lock)
 	if err != nil {
-		return nil, err
+		return lock, err
 	}
 
-	modAddr := k.ak.GetModuleAddress(types.ModuleName)
-	err = k.dk.FundCommunityPool(ctx, coins, modAddr)
+	// add lock refs into not unlocking queue
+	err = k.addLockRefs(ctx, lock)
 	if err != nil {
-		return nil, err
+		return lock, err
 	}
 
-	err = k.removeTokensFromLock(ctx, lock, coins)
+	k.SetLastLockID(ctx, lock.ID)
+	return lock, nil
+}
+
+// Lock is a utility method to lock tokens into the module account. This method includes setting the
+// lock within the state machine and increasing the value of accumulation store
+func (k Keeper) Lock(ctx sdk.Context, lock types.PeriodLock) error {
```

Comment:
What do folks think about this API change?

IMO, this should be turned into a private method and prevent confusion from using this method directly, and people should be only allowed to lock tokens with what current lock token flow allows which are 

- Add Tokens to existing locks 
- Create new lock

There aren't any other entry points of `Lock` method itself being used elsewhere as of now, thus proposing to change this into private method. 

Currently tests are failing due to this change, planning to fix it upon reviewer's decision